app/assets/javascripts/pretty-text/engines/discourse-markdown/hashtag-autocomplete.js

```diff
@@ -0,0 +1,132 @@
+// NOTE: For future maintainers, the hashtag lookup here does not take
+// into account mixed contexts -- for instance, a chat quote inside a post
+// or a post quote inside a chat message, so hashtagTypesInPriorityOrder may
+// not provide an accurate lookup for hashtags without a ::type suffix in those
+// cases if there are conflcting types of resources with the same slug.
+
+function addHashtag(buffer, matches, state) {
+  const options = state.md.options.discourse;
+  const slug = matches[1];
+  const hashtagLookup = options.hashtagLookup;
+
+  // NOTE: The lookup function is only run when cooking
+  // server-side, and will only return a single result based on the
+  // slug lookup.
+  const result =
+    hashtagLookup &&
+    hashtagLookup(
+      slug,
+      options.currentUser,
+      options.hashtagTypesInPriorityOrder
+    );
+
+  // NOTE: When changing the HTML structure here, you must also change
+  // it in the placeholder HTML code inside lib/hashtag-autocomplete, and vice-versa.
+  let token;
+  if (result) {
```

Comment:
If I read this correctly, the resulting HTML will be:

```
<a>
  <span>
    <svg />
    "text"
  </span>
</a>
```
In my book it would be better to have

```
<a>
  <svg>
  <span>text</span>
</a>
```

Reply:
Fair enough, will change